pkg/utils/watchChaosContainer.go

```diff
@@ -49,6 +50,14 @@ func GetChaosContainerStatus(experimentDetails *ExperimentDetails, clients Clien
 			}
 		}
 
+	} else if pod.Status.Phase == corev1.PodPending {
+		if waitCountLeft != 0 {
```

Comment:
This is an infinite loop in the case of the Pending state as you are initializing `waitCountLeft` this var for every iteration. Isn't it?

Reply:
initialise it from `WatchChaosContainerForCompletion`